### DISCUSSION TITLE
How to create a local user in vCenter?

### DISCUSSION THREAD

**Discussion by dzl84** [vmware/govmomi#3561] [Community Help]
Hi,

I want to use govmomi to create a local user but always got "ServerFaultCode: The request refers to an unexpected or unknown type." Below is my code. 

         var userDirectory mo.UserDirectory
	err = client.RetrieveOne(ctx, *client.ServiceContent.UserDirectory, nil, &userDirectory)
	if err != nil {
		log.Fatalf("Error retrieving UserDirectory: %v", err)
	}

	// Local user details
	newUser := types.HostAccountSpec{
		Id:                  "clouduser",            // The username of the new local user
		Password:    "<pwd>",          // The password of the new local user
		Description: "Local user ", // Description of the user

	}

	// Use methods.CreateUser to create a new local user
	req := types.CreateUser{
		This: userDirectory.Reference(),    // Reference to the UserDirectoryManager
		User: newUser.GetHostAccountSpec(), // User details
	}

	// Call AddUser method
	_, err = methods.CreateUser(ctx, client.Client, &req)
	if err != nil {
		log.Fatalf("Failed to add local user: %v", err)
	}

Please help.

**Comment by dougm** [comment]
The `CreateUser` method is for ESX hosts, not vCenter. You can see how the method is invoked by running the govc [host.account.create](https://github.com/vmware/govmomi/blob/main/govc/USAGE.md#hostaccountcreate) command.
And the source for that command: https://github.com/vmware/govmomi/blob/main/govc/host/account/create.go#L54-L58

[sso.user.create](https://github.com/vmware/govmomi/blob/main/govc/USAGE.md#ssousercreate) might be what you're looking for?
